contracts/sidechain/peripheral/SidechainClaimZap.sol

```diff
@@ -0,0 +1,245 @@
+        require(msg.sender == owner, "!auth");
+        zro = _zro;
+    }
+
+    /**
+     * @notice Returns meta data of contract
+     */
+    function getName() external pure returns (string memory) {
+        return "Sidechain ClaimZap V1.0";
+    }
+
+    /**
+     * @notice Approve spending of:
+     *          cvxCrv  -> Compounder
+     */
+    function setApprovals() external {
+        require(msg.sender == owner, "!auth");
+        _approveToken(cvxCrv, compounder);
+    }
+
+    /**
+     * @notice Allows a spender to spend a token
+     * @param _token     Token that will be spend
+     * @param _spender   Address that will be spending
+     */
+    function _approveToken(address _token, address _spender) internal {
+        IERC20(_token).safeApprove(address(_spender), 0);
+        IERC20(_token).safeApprove(address(_spender), type(uint256).max);
+    }
+
+    /**
+     * @notice Claim all the rewards
+     * @param rewardContracts        Array of addresses for LP token rewards
+     * @param extraRewardContracts   Array of addresses for extra rewards
+     * @param tokenRewardContracts   Array of addresses for token rewards e.g vlCvxExtraRewardDistribution
+     * @param tokenRewardTokens      Array of token reward addresses to use with tokenRewardContracts
+     * @param amounts                Claim rewards amounts.
+     * @param options                Claim options
+     */
+    function claimRewards(
+        address[] calldata rewardContracts,
+        address[] calldata extraRewardContracts,
+        address[] calldata tokenRewardContracts,
+        address[] calldata tokenRewardTokens,
+        ClaimRewardsAmounts calldata amounts,
+        Options calldata options
+    ) external payable {
+        require(tokenRewardContracts.length == tokenRewardTokens.length, "!parity");
+
+        //Read balances prior to reward claims only if required
+        uint256 cvxBalance;
+        uint256 cvxCrvBalance;
+        if (!options.useAllWalletFunds && _callOptions(options)) {
+            cvxBalance = IERC20(cvx).balanceOf(msg.sender);
+            cvxCrvBalance = IERC20(cvxCrv).balanceOf(msg.sender);
+        }
+
+        //claim from main curve LP pools
+        for (uint256 i = 0; i < rewardContracts.length; i++) {
+            IRewardStaking(rewardContracts[i]).getReward(msg.sender, true);
+        }
+        //claim from extra rewards
+        for (uint256 i = 0; i < extraRewardContracts.length; i++) {
+            IRewardStaking(extraRewardContracts[i]).getReward(msg.sender);
+        }
+        //claim from multi reward token contract
+        for (uint256 i = 0; i < tokenRewardContracts.length; i++) {
+            IRewardStaking(tokenRewardContracts[i]).getReward(msg.sender, tokenRewardTokens[i]);
+        }
+
+        // deposit/lock/stake
+        if (_callOptions(options)) {
+            _handleRewards(cvxBalance, cvxCrvBalance, amounts, options);
+        }
+
+        if (options.refundEth) {
```

Comment:
Is there a case when we wouldn't want to refund the ETH?

Reply:
Not that I can think of but this just seems safer